clang/test/SemaSYCL/array-kernel-param.cpp

```diff
@@ -0,0 +1,91 @@
+// RUN: %clang_cc1 -I %S/Inputs -fsycl -fsycl-is-device -ast-dump %s | FileCheck %s
+
+// This test checks that compiler generates correct kernel arguments for
+// arrays, Accessor arrays, and structs containing Accessors.
+
+#include <sycl.hpp>
+
+using namespace cl::sycl;
+
+template <typename name, typename Func>
+__attribute__((sycl_kernel)) void a_kernel(Func kernelFunc) {
+  kernelFunc();
+}
+
+int main() {
+
+  using Accessor =
+      accessor<int, 1, access::mode::read_write, access::target::global_buffer>;
+
+  Accessor acc[2];
+  int a[2];
+  struct struct_acc_t {
```

Comment:
I think the current code will not add InitListExpr for accessor array. Without this I think there will be a crash in CodeGen if struct contains more fields because InitListExpr will not match field layout. Can you check this? We might need at least an empty initializer list

Reply:
I verified this. Adding a second capture will crash in CodeGen. I think this is due to malformed AST (missing initializer for accessor array).  The same issue is causing crashes after merge in PR#1877 for accessor arrays inside structs.

Reply:
This has been fixed. Default constructor is called for each accessor.